app/workers/repository_download_worker.rb

```diff
@@ -2,7 +2,7 @@
 
 class RepositoryDownloadWorker
   include Sidekiq::Worker
-  sidekiq_options queue: :repo, lock: :until_executed
+  sidekiq_options queue: :repo, lock: :until_and_while_executing, lock_ttl: 10.minutes.to_i
```

Comment:
Will the lock expire at min(job completed time, lock_ttl), or will it always expire at lock_ttl? We could drop some legitimate close together updates if the latter

Reply:
yeah, the docs say `...how long to wait at most before considering a lock to be expired...`, so the "at most" sounds like it usually won't lock the entire TTL. (I'm having a hard time tracking down that code in the gem tho)

it looks like the P90 latency for this worker is 20-40sec, and the p99.9 reaches about 10 minutes. So if I reduce this to `1.minute.to_i` then we should guarantee locks for ~90% of these jobs?

Reply:
We could keep it at 10 and add some logging to keep tabs on the conflicts: https://github.com/mhenrixon/sidekiq-unique-jobs?tab=readme-ov-file#reflections-metrics-logging-etc